crates/agent/src/controllers/publication_status.rs

```diff
@@ -189,8 +187,9 @@ impl PendingPublication {
         tracing::info!("starting touch");
         let new_hash = new_dependency_hash.unwrap_or("None");
         let old_hash = state.live_dependency_hash.as_deref().unwrap_or("None");
-        self.details.push(format!("in response to change in dependencies, prev hash: {old_hash}, new hash: {new_hash}"));
-
+        self.details.push(format!(
+            "in response to change in dependencies, prev hash: {old_hash}, new hash: {new_hash}"
```

Comment:
I don't suppose we could include the names of the dependencies that changed easily, could we? I'm just envisioning reading a publication that says this in the future and wishing it told me _which_ dependency change was causal. Out of scope for this PR for sure, just thinking.

Reply:
To do so would require a data structure to store the `last_pub_id` of every dependency individually. It wouldn't be crazy to stuff them into the `built_spec`, and come to think of it that might have been a better approach than the one I took. But yeah, doesn't seem like the highest priority.